### PR TITLE
Handle message from broker

### PR DESCRIPTION
**Pull request by ericsciple**

### CODE REVIEW

File: src/Runner.Listener/Runner.cs [hunk 1]
```diff
@@ -469,14 +501,6 @@ private async Task<int> RunAsync(RunnerSettings settings, bool runOnce = false)
                                     Trace.Info($"Skip message deletion for cancellation message '{message.MessageId}'.");
                                 }
                             }
-                            else if (string.Equals(message.MessageType, Pipelines.HostedRunnerShutdownMessage.MessageType, StringComparison.OrdinalIgnoreCase))
```
Comment (ericsciple) [inline src/Runner.Listener/Runner.cs:472]: revert

File: src/Runner.Listener/Runner.cs [hunk 2]
```diff
@@ -510,18 +534,15 @@ private async Task<int> RunAsync(RunnerSettings settings, bool runOnce = false)
                         await jobDispatcher.ShutdownAsync();
                     }
 
-                    if (!skipSessionDeletion)
```
Comment (ericsciple) [inline src/Runner.Listener/Runner.cs:513]: revert

File: src/Runner.Listener/Runner.cs [hunk 3]
```diff
@@ -1,18 +1,22 @@
...
 using GitHub.Runner.Sdk;
-using GitHub.Services.WebApi;
-using Pipelines = GitHub.DistributedTask.Pipelines;
+using System.Linq;
+using GitHub.Runner.Listener.Check;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
```
Comment (ericsciple) [inline src/Runner.Listener/Runner.cs:19]: fix

File: src/Runner.Listener/Runner.cs [hunk 4]
```diff
@@ -458,6 +458,35 @@ private async Task<int> RunAsync(RunnerSettings settings, bool runOnce = false)
...
+                                {
+                                    var messageRef = StringUtil.ConvertFromJson<RunnerJobRequestRef>(message.Body);
+
+                                    // Create connection
+                                    var credMgr = HostContext.GetService<ICredentialManager>();
+                                    var creds = credMgr.LoadCredentials();
+
+                                    // todo: add retries https://github.com/github/actions-broker/issues/49
+                                    var runServer = HostContext.CreateService<IRunServer>();
+                                    await runServer.ConnectAsync(new Uri(settings.ServerUrl), creds);
```
Comment (TingluoHuang) [inline src/Runner.Listener/Runner.cs:479]: we are creating a new connection on every message.

Comment (ericsciple) [reply src/Runner.Listener/Runner.cs:479]: Well, we won't when we switch over to the run service 😄

Comment (TingluoHuang) [reply src/Runner.Listener/Runner.cs:479]: I am fine with the change now, but long term we might want to figure out all the URLs we need and where we get them. ex: broker URL, admin URL, runService URL.

**Review by TingluoHuang — APPROVED** [review]

**Pull request merged**